code/modules/projectiles/guns/ballistic/rifle.dm

```diff
@@ -169,6 +170,99 @@
+	rack_sound = 'sound/weapons/gun/sniper/rack.ogg'
+	must_hold_to_load = FALSE
+	mag_display = FALSE
+	empty_indicator = TRUE
+	bolt_type = BOLT_TYPE_LOCKING
+	semi_auto = FALSE
+	internal_magazine = TRUE
+	can_modify_ammo = TRUE
+	initial_caliber = CALIBER_REBAR
+	slot_flags = ITEM_SLOT_BACK|ITEM_SLOT_SUITSTORE
+	alternative_caliber = CALIBER_REBAR_FORCED
+	alternative_ammo_misfires = TRUE
+	bolt_wording = "bowstring"
+	magazine_wording = "rod"
+	cartridge_wording = "rod"
+	misfire_probability = 25
+	accepted_magazine_type = /obj/item/ammo_box/magazine/internal/boltaction/rebarxbow/normal
+	fire_sound = 'sound/items/syringeproj.ogg'
+	can_be_sawn_off = FALSE
+	tac_reloads = FALSE
+	SET_BASE_PIXEL(0, 0)
+
+/obj/item/gun/ballistic/rifle/rebarxbow/rack(mob/user = null)
+	if (bolt_locked == FALSE)
+		balloon_alert(user, "bowstring loosened")
+		playsound(src, rack_sound, rack_sound_volume, rack_sound_vary)
+		process_chamber(FALSE, FALSE, FALSE)
+		bolt_locked = TRUE
+		update_appearance()
+		return
+	drop_bolt(user)
+
+/obj/item/gun/ballistic/rifle/rebarxbow/drop_bolt(mob/user = null)
+	if(do_after(user, 3 SECONDS, target = src))
+		playsound(src, bolt_drop_sound, bolt_drop_sound_volume, FALSE)
+		if (user)
+			balloon_alert(user, "bowstring drawn")
+		chamber_round()
+		bolt_locked = FALSE
+		update_appearance()
+
+
+/obj/item/gun/ballistic/rifle/rebarxbow/can_shoot()
+	if (bolt_locked)
+		return FALSE
+	return ..()
+
+/obj/item/gun/ballistic/rifle/rebarxbow/examine(mob/user)
+	. = ..()
+	. += "The crossbow is [bolt_locked ? "not ready" : "ready"] to fire."
+
+/obj/item/gun/ballistic/rifle/rebarxbow/syndie
+	name = "Syndicate Rebar Crossbow"
+	desc = "The syndicate liked the bootleg rebar crossbow NT engineers made, so they showed what it could be if properly developed.\
+			Holds three shots, and features a built in scope. Normally uses special syndicate jagged iron bars, but can be wrenched to shoot inferior normal ones."
+	icon = 'icons/obj/weapons/guns/ballistic.dmi'
+	icon_state = "rebarxbowsyndie"
+	inhand_icon_state = "rebarxbowsyndie"
+	worn_icon_state = "rebarxbowsyndie"
+	rack_sound = 'sound/weapons/gun/sniper/rack.ogg'
+	semi_auto = FALSE
+	w_class = WEIGHT_CLASS_NORMAL
+	initial_caliber = CALIBER_REBAR_SYNDIE
+	alternative_caliber = CALIBER_REBAR_SYNDIE_NORMAL
+	alternative_ammo_misfires = FALSE
+	bolt_wording = "bowstring"
+	magazine_wording = "rod"
+	cartridge_wording = "rod"
+	accepted_magazine_type = /obj/item/ammo_box/magazine/internal/boltaction/rebarxbow/syndie
+	fire_sound = 'sound/items/syringeproj.ogg'
+	SET_BASE_PIXEL(0, 0)
+
+/obj/item/gun/ballistic/rifle/rebarxbow/syndie/drop_bolt(mob/user = null)
+	if(do_after(user, 1 SECONDS, target = src)) //shorter delay as it wasnt made by some idiot in a closet, plus it has the 3 shots
+		playsound(src, bolt_drop_sound, bolt_drop_sound_volume, FALSE)
+		if (user)
+			balloon_alert(user, "bowstring drawn")
+		chamber_round()
+		bolt_locked = FALSE
+		update_appearance()
```

Comment:
Instead of copying this whole proc just to change how long the `do_after` takes, have a var on the gun, `draw_time` that you can set differently

Reply:
ok, that sounds easier to adjust anyway.